### PR TITLE
IEP-1089 , IEP-1062: Project Creation Page updated

### PR DESCRIPTION
**Pull request by alirana01**
## Description

Remove the first page for new project so we only have the second page with project templates/examples

Fixes # ([IEP-1089](https://jira.espressif.com:8443/browse/IEP-1089))
([IEP-1062](https://jira.espressif.com:8443/browse/IEP-1062))

## Type of change

Please delete options that are not relevant.
- Imprpvement

## How has this been tested?

Try creating a new project from the project wizard. Create a project with and also without any templates. Also try to rename the examples folder to something else in esp-idf directory and see if you are still able to create a project.
Select a specific chip from the target selector in the project wizard the project should be created and the launchbar should show the target for that particular board.

Also verify that the default chip is always esp32

**Test Configuration**:
* ESP-IDF Version: any
* OS (Windows,Linux and macOS): all

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [x] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

The provided information contains a significant amount of technical details and code changes. To ensure confidentiality and focus on end-user impact, I'll need to summarize the changes based on the provided details.

Here's the revised summary:

- **New Features**
  - Enhanced project creation wizard with improved template selection and target configuration.
  - Added project target selection to streamline project setup.
  - Introduced user-friendly messages and descriptions for a better user experience.

- **Enhancements**
  - Improved project generator to handle target-specific configurations.
  - Streamlined the project creation workflow by removing unnecessary steps.

- **Refactor**
  - Renamed and restructured classes to align with the new project creation flow.
  - Updated methods to improve clarity and maintainability.

- **Bug Fixes**
  - Resolved issues related to setting the active launch target.

- **Documentation**
  - Updated user-facing strings and tooltips to enhance user experience.

- **Style**
  - Adjusted UI components for a more intuitive project setup process.

- **Tests**
  - Modified test setups to reflect changes in the project creation process.

This summary focuses on the user-facing impact of the changes while maintaining confidentiality.

### CODE REVIEW

**Review by alirana01 — COMMENTED** [review]
Self initial review. Also need to update the tests in the project

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/AbstractTemplatesSelectionPage.java [hunk 1]
```diff
@@ -85,6 +85,13 @@ public void createControl(Composite parent)
 		container.setLayoutData(new GridData(GridData.FILL_BOTH));
 
 		createAbove(container, 1);
+		if (templateElements == null || templateElements.getChildren().isEmpty())
+		{
+			Dialog.applyDialogFont(container);
+			setControl(container);
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/AbstractTemplatesSelectionPage.java:93]: Not critical here

**Review by alirana01 — COMMENTED** [review]
Reviewed after changes

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/IDFProjectGenerator.java [hunk 2]
```diff
@@ -85,8 +97,37 @@ public void generate(Map<String, Object> model, IProgressMonitor monitor) throws
...
+		ILaunchTarget launchTarget = findSuitableTargetForSelectedTargetString();
+		launchBarManager.setActiveLaunchTarget(launchTarget);
+		Logger.log("");
+	}
+	
+	private ILaunchTarget findSuitableTargetForSelectedTargetString()
+	{
+		ILaunchTargetManager launchTargetManager = UIPlugin.getService(ILaunchTargetManager.class);
+		ILaunchTarget[] targets = launchTargetManager
+				.getLaunchTargetsOfType("com.espressif.idf.launch.serial.core.serialFlashTarget"); //$NON-NLS-1$
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/IDFProjectGenerator.java:118]: we have a constant ESP_LAUNCH_TARGET_TYPE for this value in IDFLaunchConstants

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/IDFProjectGenerator.java [hunk 3]
```diff
@@ -85,8 +97,37 @@ public void generate(Map<String, Object> model, IProgressMonitor monitor) throws
...
 		// refresh to see the copied resources in the project explorer
 		project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
+		
+	}
+	
+	private void setTarget() throws CoreException
+	{
+		ILaunchTarget launchTarget = findSuitableTargetForSelectedTargetString();
+		launchBarManager.setActiveLaunchTarget(launchTarget);
+		Logger.log("");
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/IDFProjectGenerator.java:111]: why do we need empty logging here? Typo?

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/IDFProjectGenerator.java [hunk 4]
```diff
@@ -85,8 +97,37 @@ public void generate(Map<String, Object> model, IProgressMonitor monitor) throws
...
+	
+	private ILaunchTarget findSuitableTargetForSelectedTargetString()
+	{
+		ILaunchTargetManager launchTargetManager = UIPlugin.getService(ILaunchTargetManager.class);
+		ILaunchTarget[] targets = launchTargetManager
+				.getLaunchTargetsOfType("com.espressif.idf.launch.serial.core.serialFlashTarget"); //$NON-NLS-1$
+
+		for (ILaunchTarget iLaunchTarget : targets)
+		{
+			String idfTarget = iLaunchTarget.getAttribute("com.espressif.idf.launch.serial.core.idfTarget", //$NON-NLS-1$
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/IDFProjectGenerator.java:122]: we have a constant ATTR_IDF_TARGET for this value in IDFLaunchConstants

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/ImportIDFProjectWizard.java [hunk 5]
```diff
@@ -48,7 +48,7 @@ protected IGenerator getGenerator()
 		locationStr = page.getLocation();
 		boolean canCopyIntoWorkspace = page.canCopyIntoWorkspace();
 
-		IDFProjectGenerator generator = new IDFProjectGenerator(null, new File(locationStr), canCopyIntoWorkspace);
+		IDFProjectGenerator generator = new IDFProjectGenerator(null, new File(locationStr), canCopyIntoWorkspace, "esp32");
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/ImportIDFProjectWizard.java:51]: nit: we could extract "esp32" as DEFAULT_TARGET constant for better maintainability/readability

**Review by sigmaaa — COMMENTED** [review]
code LGTM. We can replace some strings with constants

**Review by alirana01 — COMMENTED** [review]
Reviewed with changes

**Comment by AndriiFilippov** [comment]
@alirana01 hi!

1)  Empty project name bad info message:

image

2) Able to create a project with a name that already exists. No validation.

3) create project A -> selecttarget (ESP32) -> build it -> create project B -> select different target -> click "Finish" -> pop-up window asks you to delete "build" folder for project A: 

image


If click "No" -> the Launch Target for project A change to target selected for project B.

image


4)  What do you think about switching Launch Configuration once switching projects ? 
I have had project A selected - once I select project B the A project Launch Config still remain.

image

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Comment by kolipakakondal** [comment]
Hi @AndriiFilippov Could you verify this again!

**Review by sigmaaa — APPROVED** [review]
LGTM

**Comment by AndriiFilippov** [comment]
@alirana01 hi! 
Tested: 
OS - Windows 10 / MacOS 
ESP-IDF: v5.1.1 

1) Project Name: empty / duplicate / special symbols.  👍
Question: able to create project with name ->  .,
And this project is not visible in the Project Explorer then. 

2) Project location: able to create external workspace project. can't create more then 1 project in the same external workspace  👍
But is there PATH validation ?  It seems like you can write any path starting with '/'. 

Screenshot 2023-11-15 at 22 27 59


3) Template Selection / Filter   👍

**Comment by kolipakakondal** [comment]
> @alirana01 hi! Tested: OS - Windows 10 / MacOS ESP-IDF: v5.1.1
> 
> 1. Project Name: empty / duplicate / special symbols.  👍
>    Question: able to create project with name ->  .,
>    And this project is not visible in the Project Explorer then.

Looks like this issue is there with the Eclipse C/C++ project as well(I'm able to create a project with ., but it appeared in in the project explorer), we can fix this. Please create a seperate ticket.

> 2. Project location: able to create external workspace project. can't create more then 1 project in the same external workspace  👍

> 
Probably you would have confused with the "Location" you provide when you uncheck "use default location", what is captured is the proejct location not the workspace location hence it won't allow duplicate projects with in the same location. BTW, project name can be different from the folder name. For example, I could provide project name as "proj1" and location could be "/Users/kondal/esp/test", in this case "test" is a project folder name and .project will have a project attribute as "proj1"

**Comment by alirana01** [comment]
@AndriiFilippov there can be a path starting with **/** on mac and linux so shouldn't be a problem

**Comment by alirana01** [comment]
> > @alirana01 hi! Tested: OS - Windows 10 / MacOS ESP-IDF: v5.1.1
> > 
> > 1. Project Name: empty / duplicate / special symbols.  👍
> >    Question: able to create project with name ->  .,
> >    And this project is not visible in the Project Explorer then.
> 
> Looks like this issue is there with the Eclipse C/C++ project as well(I'm able to create a project with ., but it appeared in in the project explorer), we can fix this. Please create a seperate ticket.
> 
> > 2. Project location: able to create external workspace project. can't create more then 1 project in the same external workspace  👍
> 
> > 
> 
> Probably you would have confused with the "Location" you provide when you uncheck "use default location", what is captured is the proejct location not the workspace location hence it won't allow duplicate projects with in the same location. BTW, project name can be different from the folder name. For example, I could provide project name as "proj1" and location could be "/Users/kondal/esp/test", in this case "test" is a project folder name and .project will have a project attribute as "proj1"

@kolipakakondal @AndriiFilippov 
I think a project starting with . is considered a hidden directory as per linux and mac and I think eclipse will ignore those directories specifically so not sure if this is supposed to be a bug

**Pull request merged**